R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,717 @@
+   "source": [
+    "Save the ROIs back to OMERO"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# addROIs creates an ROI for each entry in the dataframe specified by the 'coords' parameter\n",
+    "addROIs(image, coords = rois)\n",
+    "print(\"ROIs successfully created.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Open the image in the full viewer in OMERO.web and check the ROIs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Store the features table"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Attach the whole 'features' dataframe to the image:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 'attachDataframe' directly attaches an R dataframe to an OMERO image (project, dataset, etc.)\n",
+    "# ('invisible' not strictly necessary, just suppresses some unnecessary output)\n",
+    "invisible(attachDataframe(image, features, \"ROI features\"))\n",
+    "print(\"Data frame successfully attached.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The dataframe is now attached to the image as an [HDF](https://en.wikipedia.org/wiki/Hierarchical_Data_Format) table with file name 'ROI features'. You can download and open it with software like HDF Compass, load it into python scripts using 'h5py' library, etc. or simply load it directly as R dataframe again."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Alternatively attach it as CSV file:\n",
+    "\n",
+    "- Use R's built-in write.csv method to create a csv file\n",
+    "- And attach that to the image using attachFile function from the romero.gateway"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "csvFile <- \"/tmp/ROI_features.csv\"\n",
+    "write.csv(features, file = csvFile)\n",
+    "fileannotation <- attachFile(image, csvFile)\n",
+    "print(\"CSV file successfully attached.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Automate"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Tasks:\n",
+    "\n",
+    "- Segment each image of a dataset.\n",
+    "- For visualisation create an ROI for each of the objects detected.\n",
+    "- For further analysis only take the largest object per image into account (assumption: this is a centriole). Parameters to evaluate for this object are area, perimeter and diameter\n",
+    "\n",
+    "Note: We need to be able to specify the channel which is analyzed by its name. Only one channel in each image is relevant for our analysis. The relevant channel is identified by its name.\n",
```

Comment:
I would simply remove the words ``...which is analyzed by its name.`` You are explaining it well enough in the following sentences, and with better wording even.

Reply:
Add a sentence:

``The name must match the name of the containing dataset.``